lib/http.js

```diff
@@ -1344,6 +1344,22 @@ Agent.prototype.removeSocket = function(s, name, host, port, localAddress) {
 var globalAgent = new Agent();
 exports.globalAgent = globalAgent;
 
+function check_is_token(val) {
+  if (!val || typeof val !== 'string')
+    return false;
+  for (var n = 0, l = val.length; n < l; n++) {
+    var c = val.charCodeAt(n);
+    if (c <= 32 || c == 34 ||
+        c == 44 || c == 47 ||
+        c == 123 || c == 125 ||
+        c == 40 || c == 41 ||
+        (c >= 58 && c <= 64) ||
+        (c >= 91 && c <= 93) || 
+        c >= 128)
+      return false;
+  }
+  return true;
```

Comment:
Since all callers are just going to take false and `throw` I would advocate this method should just `throw`, then all our state is preserved.

Also as far as the error message, I would like to improve it -- ideally the message would be something like: `"Invalid character in HTTP method at %d for %s", n, val`

Then on the actual error object that's constructed attach the same information.

If we're against throwing in this method, I would advocate returning null or falsey in the happy case, and returning the as described error object in the case where we should throw.